vendor/github.com/benburkert/dns/message.go

```diff
@@ -0,0 +1,975 @@
+	TypeCAA   Type = 257 // [RFC6844] Certification Authority Restriction
+
+	TypeANY Type = 0
+
+	// DNS CLASSes
+	ClassIN  Class = 1   // [RFC1035] Internet (IN)
+	ClassCH  Class = 3   // [] Chaos (CH)
+	ClassHS  Class = 4   // [] Hesiod (HS)
+	ClassANY Class = 255 // [RFC1035] QCLASS * (ANY)
+
+	// DNS RCODEs
+	NoError  RCode = 0 // [RFC1035] No Error
+	FormErr  RCode = 1 // [RFC1035] Format Error
+	ServFail RCode = 2 // [RFC1035] Server Failure
+	NXDomain RCode = 3 // [RFC1035] Non-Existent Domain
+	NotImp   RCode = 4 // [RFC1035] Not Implemented
+	Refused  RCode = 5 // [RFC1035] Query Refused
+
+	maxPacketLen = 512
+)
+
+// NewRecordByType returns a new instance of a Record for a Type.
+var NewRecordByType = map[Type]func() Record{
+	TypeA:     func() Record { return new(A) },
+	TypeNS:    func() Record { return new(NS) },
+	TypeCNAME: func() Record { return new(CNAME) },
+	TypeSOA:   func() Record { return new(SOA) },
+	TypePTR:   func() Record { return new(PTR) },
+	TypeMX:    func() Record { return new(MX) },
+	TypeTXT:   func() Record { return new(TXT) },
+	TypeAAAA:  func() Record { return new(AAAA) },
+	TypeSRV:   func() Record { return new(SRV) },
+	TypeDNAME: func() Record { return new(DNAME) },
+	TypeOPT:   func() Record { return new(OPT) },
+	TypeCAA:   func() Record { return new(CAA) },
+}
+
+var (
+	// ErrNotStarted indicates that the prerequisite information isn't
+	// available yet because the previous records haven't been appropriately
+	// parsed or skipped.
+	ErrNotStarted = errors.New("parsing of this type isn't available yet")
+
+	// ErrSectionDone indicated that all records in the section have been
+	// parsed.
+	ErrSectionDone = errors.New("parsing of this section has completed")
+
+	errBaseLen            = errors.New("insufficient data for base length type")
+	errCalcLen            = errors.New("insufficient data for calculated length type")
+	errReserved           = errors.New("segment prefix is reserved")
+	errPtrCycle           = errors.New("pointer cycle")
+	errInvalidFQDN        = errors.New("invalid FQDN")
+	errInvalidPtr         = errors.New("invalid pointer")
+	errResourceLen        = errors.New("insufficient data for resource body length")
+	errSegTooLong         = errors.New("segment length too long")
+	errZeroSegLen         = errors.New("zero length segment")
+	errResTooLong         = errors.New("resource length too long")
+	errTooManyQuestions   = errors.New("too many Questions to pack (>65535)")
+	errTooManyAnswers     = errors.New("too many Answers to pack (>65535)")
+	errTooManyAuthorities = errors.New("too many Authorities to pack (>65535)")
+	errTooManyAdditionals = errors.New("too many Additionals to pack (>65535)")
+	errFieldOverflow      = errors.New("value too large for packed field")
+	errUnknownType        = errors.New("unknown resource type")
+)
+
+// Message is a DNS message.
+type Message struct {
```

Comment:
I don't think we should be using this message implementation. It seems to be an old copy of the one [currently used](https://github.com/golang/net/blob/master/dns/dnsmessage/message.go) in stdlib, and we should rely on that one instead. This probably means that we can't use `benburkert/dns` as a library and will need to re-implement what we need ourselves.